runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -110,6 +110,46 @@ public void bootstrapNewNode(String endpoint)
         log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
     }
 
+    /**
+     * Best effort attempt to removes a node from the layout. The remove
+     * is not guaranteed and the caller is responsible for retrying.
+     *
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(Layout currentLayout, String endpoint) throws QuorumUnreachableException,
+            CloneNotSupportedException,
+            LayoutModificationException, OutrankedException {
+
+        LayoutBuilder builder = new LayoutBuilder(currentLayout);
+        Layout newLayout = builder.removeLayoutServer(endpoint)
+                .removeSequencerServer(endpoint)
+                .removeLogunitServer(endpoint)
+                .removeUnResponsiveServer(endpoint)
+                .setEpoch(currentLayout.getEpoch() + 1)
+                .build();
+
+        // Seal after constructing the layout, so that the system
+        // isn't blocked if the builder throws an exception
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        newLayout.setRuntime(runtime);
+        attemptConsensus(newLayout);
+
+        runtime.invalidateLayout();
+
+        try {
+            reconfigureSequencerServers(currentLayout, newLayout, false);
+        } catch (InterruptedException ie) {
+            log.error("removeNode: Bootstrapping sequencer failed due to exception : ", ie);
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(ie);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)